src/components/icons/ChainLogo.tsx

```diff
@@ -1,29 +1,49 @@
 import Image from 'next/image';
-import { ComponentProps, useMemo } from 'react';
+import { useMemo } from 'react';
 
 import { ChainLogo as ChainLogoInner } from '@hyperlane-xyz/widgets';
 
-import { getChainDisplayName, tryGetChainMetadata } from '../../features/chains/utils';
+import { getRegistry } from '../../context/context';
+import { tryGetChainMetadata } from '../../features/chains/utils';
 
-export function ChainLogo(props: ComponentProps<typeof ChainLogoInner>) {
-  const { chainName, ...rest } = props;
-  const { chainId, chainDisplayName, icon } = useMemo(() => {
-    if (!chainName) return {};
-    const chainDisplayName = getChainDisplayName(chainName);
-    const chainMetadata = tryGetChainMetadata(chainName);
-    const chainId = chainMetadata?.chainId;
+export function ChainLogo({
+  chainId,
+  chainName,
+  background,
+  size,
+}: {
+  chainId?: ChainId;
+  chainName?: string;
+  background?: boolean;
+  size?: number;
+}) {
+  const registry = getRegistry();
+  const { name, icon } = useMemo(() => {
+    const chainNameOrId = chainName || chainId;
+    if (!chainNameOrId) return { name: '' };
+    const chainMetadata = tryGetChainMetadata(chainNameOrId);
+    const name = chainMetadata?.name || '';
     const logoUri = chainMetadata?.logoURI;
     const icon = logoUri
       ? (props: { width: number; height: number; title?: string }) => (
           <Image src={logoUri} alt="" {...props} />
         )
       : undefined;
     return {
-      chainId,
-      chainDisplayName,
+      name,
       icon,
     };
-  }, [chainName]);
+  }, [chainName, chainId]);
 
-  return <ChainLogoInner {...rest} chainId={chainId} chainName={chainDisplayName} icon={icon} />;
+  return (
+    <ChainLogoInner
+      chainName={name}
+      // TODO
+      // @ts-ignore update registry version in widget lib
```

Comment:
should we maybe move the widgets lib into the monorepo? seems a little cumbersome atm to juggle across the repos

Reply:
Yeah we also decided that we would remember? There's a ticket but it was de-prioritized because now that there aren't hard-coded chain logos in it, it doesn't need to change as often. https://github.com/hyperlane-xyz/issues/issues/1111